src/pages/test-report/components/uncouple-recouple/uncouple-recouple.ts

```diff
@@ -0,0 +1,193 @@
+  selectedUncoupleRecouple$: Observable<boolean>;
+  uncoupleRecoupleOutcome$: Observable<CompetencyOutcome>;
+}
+
+@Component({
+  selector: 'uncouple-recouple',
+  templateUrl: 'uncouple-recouple.html',
+})
+
+export class UncoupleRecoupleComponent implements OnInit, OnDestroy {
+  componentState: UncoupleRecoupleComponentState;
+  subscription: Subscription;
+
+  isRemoveFaultMode: boolean = false;
+  isSeriousMode: boolean = false;
+  isDangerousMode: boolean = false;
+
+  selectedUncoupleRecouple: boolean = false;
+  uncoupleRecoupleOutcome: CompetencyOutcome;
+  merged$: Observable<boolean | CompetencyOutcome>;
+
+  constructor(private store$: Store<StoreModel>) {
+  }
+
+  ngOnInit(): void {
+    const currentTest$ = this.store$.pipe(
+      select(getTests),
+      select(getCurrentTest),
+    );
+
+    this.componentState = {
+      isRemoveFaultMode$: this.store$.pipe(
+        select(getTestReportState),
+        select(isRemoveFaultMode)),
+      isSeriousMode$: this.store$.pipe(
+        select(getTestReportState),
+        select(isSeriousMode)),
+      isDangerousMode$: this.store$.pipe(
+        select(getTestReportState),
+        select(isDangerousMode)),
+      selectedUncoupleRecouple$: currentTest$.pipe(
+        select(getTestData),
+        select(testData => get(testData, 'uncoupleRecouple.selected')),
+      ),
+      uncoupleRecoupleOutcome$: currentTest$.pipe(
+        select(getTestData),
+        select(testData => get(testData, 'uncoupleRecouple.fault')),
+      ),
+    };
+
+    const {
+      isRemoveFaultMode$,
+      isSeriousMode$,
+      isDangerousMode$,
+      selectedUncoupleRecouple$,
+      uncoupleRecoupleOutcome$,
+    } = this.componentState;
+
+    this.subscription = merge(
+      isRemoveFaultMode$.pipe(map(toggle => this.isRemoveFaultMode = toggle)),
+      isSeriousMode$.pipe(map(toggle => this.isSeriousMode = toggle)),
+      isDangerousMode$.pipe(map(toggle => this.isDangerousMode = toggle)),
+      selectedUncoupleRecouple$.pipe(map(value => this.selectedUncoupleRecouple = value)),
+      uncoupleRecoupleOutcome$.pipe(map(outcome => this.uncoupleRecoupleOutcome = outcome)),
+    ).subscribe();
+
+  }
+
+  ngOnDestroy(): void {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+
+  onTap = () => {
+    this.addOrRemoveFault();
+  }
+
+  onPress = () => {
+    this.addOrRemoveFault(true);
+  }
+
+  canButtonRipple = (): boolean => {
+    if (this.isRemoveFaultMode) {
+      if (this.hasDangerousFault() && this.isDangerousMode) {
+        return true;
+      }
+
+      if (this.hasSeriousFault() && this.isSeriousMode) {
+        return true;
+      }
+
+      if (!this.isSeriousMode && !this.isDangerousMode && this.faultCount() > 0) {
+        return true;
+      }
+      return false;
+    }
+    return !(this.hasDangerousFault() || this.hasSeriousFault() || this.faultCount() > 0);
+  }
+
+  toggleUncoupleRecouple = (): void => {
```

Comment:
Any unit tests for these functions?

Reply:
will be added in a separate PR (MES-4189)